spec/ruby/optional/capi/module_spec.rb

```diff
@@ -322,6 +322,11 @@ def ruby_test_method
       @class.should_not have_instance_method(:ruby_test_method)
     end
 
+    it "undefines private methods also" do
+      @m.rb_undef_method @class, "initialize"
```

Comment:
This looks good, however it sounds a bit dangerous to undefine `BasicObject#initialize`, even though if the implementation is correct this would just add an `undef` method entry in `@class`. If the implementation is incorrect, it might lead to hard-to-debug problems.

Could you add an extra `private` method in `@class` and test on that?

Reply:
:+1: I modified the test to undefine `initialize_copy` instead, which is what the `openssl` gem already does, and to test by calling `dup`, which is what `OpenSSL::TestSSL` and `OpenSSL::TestX509Store` already test.

Reply:
Great, thanks, I like specs driven by real code usages :)